smartlearner/optimizers/adagrad.py

```diff
@@ -54,3 +58,16 @@ def _get_directions(self):
             directions[param] = (self.lr/root_sum_squared) * direction
 
         return directions
+
+    def _save(self, path):
+        state = {"version": 1}
+        for k, param in self.parameters.items():
+            state[k] = param.get_value()
+
+        np.savez(pjoin(path, 'adagrad.npz'), **state)
```

Comment:
`_save` should return a `state` dictionary that `save` will handle. No point calling `np.savez` all the time with maybe different conventions. `self.__name__` might be useful?